lib/dash/segment_list.js

```diff
@@ -243,7 +240,9 @@ shaka.dash.SegmentList = class {
         endTime = startTime + periodDuration;
       }
 
-      const getUris = () => mediaUri;
+      const getUris = () => {
```

Comment:
`getUris()` is called every time `shaka.media.SegmentIndex.current()` is called, which happens every update of the streaming engine I believe. That's alongside other calls at other times.
If `resolveUris()` is taking a significant amount of time, we should probably cache the result in here to make sure that we don't have to call it again when this method is called again.

Reply:
Added!